File: be/src/main/java/sg/flow/services/TransactionHistoryServices/TransactionHistoryService.java
package sg.flow.services.TransactionHistoryServices;

import java.time.LocalDate;

import jakarta.websocket.SendResult;
import sg.flow.models.transaction.TransactionHistoryDetail;
import sg.flow.models.transaction.TransactionHistoryList;
import sg.flow.models.transfer.TransferRecepient;
import sg.flow.models.transfer.TransferRequestBody;

public interface TransactionHistoryService {

    TransactionHistoryList getMonthlyTransaction(int userId, int year, int month);

    TransactionHistoryList getDailyTransaction(int userId, LocalDate date);

    TransactionHistoryDetail getTransactionDetails(int userId, String transaction_id);

    TransferRecepient getRelevantRecepient(String keyword);

    SendResult sendTransaction(String recepientId, Double amount);

    SendResult sendTransaction(TransferRequestBody sendRequestBody);

    TransactionHistoryList getLast30DaysHistoryList(int userId);

    TransactionHistoryList getTransactionWithinRange(int userId, LocalDate startDate, LocalDate endDate);

}

File: be/src/main/kotlin/sg/flow/services/TransactionHistoryServices/TransactionHistoryService.kt
package sg.flow.services.TransactionHistoryServices

import sg.flow.entities.RecurringSpendingMonthly
import sg.flow.models.transaction.TransactionHistoryDetail
import sg.flow.models.transaction.TransactionHistoryList
import sg.flow.models.transfer.TransferRecepient
import java.time.LocalDate

interface TransactionHistoryService {
    suspend fun getMonthlyTransaction(userId: Int, year: Int, month: Int): TransactionHistoryList
    suspend fun getDailyTransaction(userId: Int, date: LocalDate): TransactionHistoryList
    suspend fun getTransactionDetails(userId: Int, transactionId: String): TransactionHistoryDetail
    suspend fun getRelevantRecepient(keyword: String): TransferRecepient
    suspend fun getLast30DaysHistoryList(userId: Int): TransactionHistoryList
    suspend fun getTransactionWithinRange(userId: Int, startDate: LocalDate, endDate: LocalDate): TransactionHistoryList
    suspend fun getProcessedTransactionsForTransactionIds(userId: Int, transactionIds: List<String>): TransactionHistoryList
    suspend fun getRecurringTransactionAnalysisResult(userId: Int): List<RecurringSpendingMonthly>
    suspend fun setTransactionCategory(userId: Int, transactionId: String, category: String): Boolean
} 